### PR TITLE
Started capturing the average usd rates for a month.

### PR DESCRIPTION
**Pull request by Himanshunegi0332**
Targets #2380 

### Description
Started capturing the currency, average usd rates for a month in database for future refernce.

### Checklist:
- [X] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Invoice/Services/InvoiceService.php [hunk 1]
```diff
@@ -157,6 +158,12 @@ public function store($data)
         $invoice = Invoice::create($data);
         $this->saveInvoiceFile($invoice, $data['invoice_file']);
         $this->setInvoiceNumber($invoice, $data['sent_on']);
+        $currentDate = Carbon::now();
+        CurrencyAvgRates::create([
+            'currency' =>  $data['currency'],
+            'captured_for' => $currentDate,
+            'avg_rate' => $this->currencyService()->getCurrentRatesInINR(),
+        ]);
```
Comment (pankaj-ag) [inline Modules/Invoice/Services/InvoiceService.php:166]:
```suggestion
       
```
@Himanshunegi0332 I think there is some confusion. 

We can't add this code here.
We need to do that via cron or some API calls.

For now, let's create a table only and we can fill it from the backend.

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: Modules/Invoice/Entities/CurrencyAvgRates.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Modules\Invoice\Entities;
+
+use Illuminate\Database\Eloquent\Model;
+
+class CurrencyAvgRates extends Model
```
[the remark this answers is not in the document]
Comment (Himanshunegi0332) [reply Modules/Invoice/Entities/CurrencyAvgRates.php:7]: Ok sir

File: Modules/Invoice/Entities/CurrencyAvgRates.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+{
+    protected $table = 'currency_avg_rate';
+    protected $fillable = ['currency', 'captured_for', 'avg_rate', 'created_at', 'updated_at'];
```
[the remark this answers is not in the document]
Comment (Himanshunegi0332) [reply Modules/Invoice/Entities/CurrencyAvgRates.php:10]: Ok sor

File: Modules/Invoice/Services/InvoiceService.php (hunk 1 above)
Comment (Himanshunegi0332) [reply Modules/Invoice/Services/InvoiceService.php]: Ok for now I have created the table and model.

**Review by pankaj-ag — APPROVED** [review]

**Pull request merged**